File: frontend/lib/app/models/permission.dart
import 'package:json_annotation/json_annotation.dart';

part 'permission.g.dart';

@JsonSerializable()
class Permission {
  final int id;
  @JsonKey(defaultValue: '')
  final String name;
  @JsonKey(name: 'display_name', defaultValue: '')
  final String displayName;
  final String? description;
  final String? resource;
  final String? action;
  @JsonKey(name: 'is_active', defaultValue: true)
  final bool isActive;
  @JsonKey(name: 'created_at')
  final DateTime? createdAt;
  @JsonKey(name: 'updated_at')
  final DateTime? updatedAt;

  Permission({
    required this.id,
    this.name = '',
    this.displayName = '',
    this.description,
    this.resource,
    this.action,
    this.isActive = true,
    this.createdAt,
    this.updatedAt,
  });

  factory Permission.fromJson(Map<String, dynamic> json) =>
      _$PermissionFromJson(json);

  Map<String, dynamic> toJson() => _$PermissionToJson(this);
}

File: backend/app/schemas/permission.py
# app/schemas/permission.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class PermissionBase(BaseModel):
    name: str = Field(..., max_length=100)
    display_name: str = Field(..., max_length=100)
    description: Optional[str] = None
    resource: Optional[str] = Field(None, max_length=50)
    action: Optional[str] = Field(None, max_length=50)

class PermissionCreate(PermissionBase):
    is_active: bool = True

class PermissionUpdate(BaseModel):
    display_name: Optional[str] = Field(None, max_length=100)
    description: Optional[str] = None
    resource: Optional[str] = Field(None, max_length=50)
    action: Optional[str] = Field(None, max_length=50)
    is_active: Optional[bool] = None

class PermissionResponse(PermissionBase):
    id: int
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
